torch2jax/__init__.py

```diff
@@ -879,6 +879,40 @@ def dropout(input, p=0.5, training=True, inplace=False):
     return _v(input)
 
 
+@implements(torch.nn.functional.embedding)
+def embedding(input, weight, padding_idx=None, max_norm=None, norm_type=2.0, scale_grad_by_freq=False, sparse=False):
+  assert max_norm is None, "torch did an inplace set of weight, which is not possible for jax"
+  assert not sparse, "TODO: implement sparse"
+  input = _v(input)
+  weight = _v(weight)
+
+  if padding_idx is not None:
+    weight = weight.at[padding_idx].set(jax.lax.stop_gradient(weight[padding_idx]))
+
+  if scale_grad_by_freq:
+
+    @jax.custom_jvp
+    def f(weight):
+      return weight[input]
+
+    @f.defjvp
+    def f_jvp(primals, tangents):
+      w = primals[0]
+      w_dot = tangents[0]
+      primal_out = f(w)
+      size = min(w.shape[0], input.size)
+      indices, counts = jnp.unique_counts(input, size=size, fill_value=w.shape[0])
+      inv_freq = 1.0 / counts
+      slicing = (...,) + (None,) * (weight.ndim - 1)
+      inv_freq = inv_freq[slicing]
+      w_dot = w_dot.at[indices].multiply(inv_freq, indices_are_sorted=True, mode="drop")
```

Comment:
```suggestion
      w_dot = w_dot.at[indices].multiply(inv_freq, mode="drop")
```
Although this is clever and it does seem to hold, AFAICT the spec for `jnp.unique_counts` makes no guarantee that `indices` will be sorted (https://docs.jax.dev/en/latest/_autosummary/jax.numpy.unique_counts.html). Feel free to correct me if I'm missing something.

Reply:
yeah on one hand i know that `unique_counts` is currently implemented this way, but i'm paranoid and so i don't feel comfortable unless they explicitly specify that in the documentation. also it may be worth noting that `sorted=True` is ignored by JAX's implementation of `jnp.unique`: https://docs.jax.dev/en/latest/_autosummary/jax.numpy.unique.html